core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -935,8 +935,9 @@ private[group] class GroupCoordinator(
           producerId,
           producerEpoch,
           RecordBatch.NO_SEQUENCE,
-          requestLocal,
-          postVerificationCallback
+          // Wrap the callback to be handled on an arbitrary request handler thread
+          // when transaction verification is complete.
+          KafkaRequestHandler.wrapAsyncCallback(postVerificationCallback, requestLocal)
```

Comment:
It's interesting to note (I don't think we need to change anything) is that now we'll have a production code path (and not just unit test) where we can call wrapped callback on the same request thread and we'll go through the optimized code path where we call the callback directly.

Reply:
Why are we calling this here? I thought we wanted to avoid this wrap here and only do it for produce requests.

Reply:
We still need this in the old coordinator.

Reply:
Oh sorry. I guess I was just confused I didn't see it in the replica manager flow. (for produce)